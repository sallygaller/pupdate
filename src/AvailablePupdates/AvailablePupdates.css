.AvailablePupdates h2 {
    text-align: center;
}