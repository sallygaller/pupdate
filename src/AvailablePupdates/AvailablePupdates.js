import React from "react";
import Pupdate from "../Pupdate/Pupdate";
import Accordion from "../Accordion/Accordion";
import { API_ENDPOINT } from "../config";
import TokenService from "../services/token-service";
import "./AvailablePupdates.css";

class AvailablePupdates extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      availablePupdates: [],
      error: null,
    };
  }

  componentDidMount() {
    Promise.all([
      // get all pupdates
      fetch(API_ENDPOINT + `/pupdates`, {
        headers: {
          authorization: `bearer ${TokenService.getAuthToken()}`,
        },
      }),
      // get pupdate rsvps for logged in user
      fetch(API_ENDPOINT + `/pupdate-rsvp/user`, {
        headers: {
          authorization: `bearer ${TokenService.getAuthToken()}`,
        },
      }),
      // get pupdates that user has organized
      fetch(API_ENDPOINT + `/pupdates/user`, {
        headers: {
          authorization: `bearer ${TokenService.getAuthToken()}`,
        },
      }),
    ])
      .then(([res1, res2, res3]) =>
        Promise.all([res1.json(), res2.json(), res3.json()])
      )
      .then(([responseData1, responseData2, responseData3]) => {
        let finalArray = [];
        while (responseData2.length || responseData3.length) {
          let value;
          if (
            responseData3.length === 0 ||
            (responseData2.length &&
              responseData2[0].pupdate < responseData3[0].id)
          ) {
            value = responseData2[0].pupdate;
            responseData2.splice(0, 1);
          } else if (
            responseData2.length === 0 ||
            responseData2[0].pupdate > responseData3[0].id
          ) {
            value = responseData3[0].id;
            responseData3.splice(0, 1);
          } else {
            value = responseData2[0].pupdate;
            responseData2.splice(0, 1);
            responseData3.splice(0, 1);
          }
          finalArray.push(value);
        }
        // filter response data to see which pupdates user has RSVP'd to
        this.setState({
          availablePupdates: responseData1.filter((pupdate) => {
            return !finalArray.find((p) => p === pupdate.id);
          }),
        });
      })
      .catch((error) => {
        console.error({ error });
      });
  }

  render() {
    return (
      <div className="AvailablePupdates">
        <section>
          <div>
            <Accordion
              title="Available Pupdates"
              pupdates={this.state.availablePupdates}
              userOrganized="false"
              userAttending="false"
              availablePupdate="true"
            />
            {/* <ul>
              {this.state.availablePupdates.map((availablePupdate) => (
                <li key={availablePupdate.id}>
                  <Pupdate pupdate={availablePupdate} availablePupdate="true" />
                </li>
              ))}
            </ul> */}
          </div>
        </section>
      </div>
    );
  }
}

export default AvailablePupdates;
