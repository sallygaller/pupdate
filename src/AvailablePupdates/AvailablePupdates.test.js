import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import AvailablePupdates from "./AvailablePupdates";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <BrowserRouter>
      <AvailablePupdates />
    </BrowserRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it("renders the UI as expected", () => {
  const tree = renderer
    .create(<AvailablePupdates name="AvailablePupdates" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
