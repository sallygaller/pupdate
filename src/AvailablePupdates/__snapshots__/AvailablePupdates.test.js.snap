// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="AvailablePupdates"
>
  <section>
    <div>
      <div
        className="main Accordion"
      >
        <div
          className="section"
        >
          <button>
            toggle
          </button>
          <div
            className="sectionhead"
            onClick={[Function]}
          >
            <h3>
              Available Pupdates
            </h3>
          </div>
          <div
            className="hidden"
          >
            <div
              className="article"
            >
               
              <p>
                No pupdates here! Why don't you
                 
                <a
                  href="/new-pupdate"
                >
                  create one?
                </a>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
